easybuild/easyblocks/m/modeller.py

```diff
@@ -47,57 +97,66 @@ def build_step(self):
-
-        qa = {
-            # installer will autodetect the right arch. [3] = x86_64
-            'Select the type of your computer from the list above [3]:': '',
-            "[%s]:" % default_install_path: self.installdir,
-            'http://salilab.org/modeller/registration.html:': self.cfg["key"],
-            'https://salilab.org/modeller/registration.html:': self.cfg["key"],
-            'Press <Enter> to begin the installation:': '',
-            'Press <Enter> to continue:': ''
-        }
-
-        run_cmd_qa(cmd, qa, log_all=True, simple=True)
+        run_cmd_qa(cmd, self.qa, log_all=True, simple=True)
+        # Determine lib/arch_ according to modeller's architecure naming scheme. After running the installer for the
+        # first time, there should only be one subdirectory in lib, e.g. x86_64-intel8. Save this value for later
+        # multi_dep interations, as lib will already be populated.
+        if not self.arch_:
+            self.arch_ = os.listdir(os.path.join(self.installdir, 'lib'))[0]
+        # _modeller.so is provided for different Python versions, namely 2.5, 3.0, 3.2, and >=3.3
+        # (still the case for Mod. 10.2)
+        # We link the _modeller.so and %(installdir)s/modlib into %(installdir)s/lib/python%(pyshortver)s/site-packages
+        # in order to allow multi_deps python
+        py_api_dirname = 'python2.5'
+        pyshortver = '.'.join(get_software_version('Python').split('.')[:2])
+        python_path = os.path.join(self.installdir, 'lib', 'python%s', 'site-packages') % pyshortver
+        if python_looseversion >= LooseVersion('3.0'):
+            py_api_dirname = 'python3.0'
+        if python_looseversion >= LooseVersion('3.2'):
+            py_api_dirname = 'python3.2'
+        if python_looseversion >= LooseVersion('3.3'):
+            py_api_dirname = 'python3.3'
+        mkdir(python_path, True)
+        os.listdir(python_path)
+        for src in os.listdir(os.path.join(self.installdir, 'modlib')):
+            os.symlink(os.path.join('..', '..', '..', 'modlib', src), os.path.join(python_path, src))
+        for src in os.listdir(os.path.join(self.installdir, 'lib', self.arch_, py_api_dirname)):
+            os.symlink(os.path.join('..', '..', self.arch_, py_api_dirname, src), os.path.join(python_path, src))
+        # link all shared libraries from the architecture specific directory (e.g. x86_64-intel8) into lib. Exclude
+        # _modeller.so, as this library is for Python <2.5.
+        for src in [
+            f for f in os.listdir(os.path.join(self.installdir, 'lib', self.arch_))
+            if os.path.isfile(os.path.join(self.installdir, 'lib', self.arch_, f))
+            and f != '_modeller.'+get_shared_lib_ext()]:
+                dst = os.path.join(self.installdir, 'lib', src)
+                if not os.path.exists(dst):
+                    os.symlink(os.path.join(self.arch_, src), dst)
```

Comment:
@ThomasHoffmann77 the hound (flake8) still complains about the formatting:
```suggestion
        for src in [
            f for f in os.listdir(os.path.join(self.installdir, 'lib', self.arch_))
            if os.path.isfile(os.path.join(self.installdir, 'lib', self.arch_, f))
                and f != '_modeller.'+get_shared_lib_ext()
        ]:
            dst = os.path.join(self.installdir, 'lib', src)
            if not os.path.exists(dst):
                os.symlink(os.path.join(self.arch_, src), dst)
```